### PR TITLE
shutdown on SIGTERM as well as SIGINT

### PR DESCRIPTION
**Pull request by heckj**
This adds a signal handler for SIGTERM to "politely" shutdown the server when the process receives that signal. This is a partial (and likely incomplete) addition to https://github.com/vapor/vapor/issues/1889

This is my first contribution to Vapor, and I'm not what I'd call deeply conversant with Swift, so feedback or changes desired are very welcome. I didn't see any existing tests that would exercise the signal handling code to amend into this PR.

I wasn't comfortable enough with all the changes recommended in #1889 to set up a more extensive PR for that issue.

### Checklist

- [x] Circle CI is passing (code compiles and passes tests).
- [x] There are no breaking changes to public API.
- [x] New test cases have been added where appropriate.
- [x] All new code has been commented with doc blocks `///`.

### CODE REVIEW

File: Sources/Vapor/Commands/ServeCommand.swift [hunk 1]
```diff
@@ -70,8 +70,13 @@ public final class ServeCommand: Command {
             _ = self.runningServer?.close()
             signalSource.cancel()
         }
-        signal(SIGINT, SIG_IGN)
-        signalSource.resume()
```
Comment (tanner0101) [inline Sources/Vapor/Commands/ServeCommand.swift]: I think we need to keep these two lines to continue capturing `SIGINT`. You can test if `SIGINT` is working by doing `ctrl+c` (`^C`) in your terminal while the HTTP server is running.

File: Sources/Vapor/Commands/ServeCommand.swift [hunk 2]
```diff
@@ -70,8 +70,13 @@ public final class ServeCommand: Command {
...
+        let signalSourceTERM = DispatchSource.makeSignalSource(signal: SIGTERM, queue: signalQueue)
+        signalSourceTERM.setEventHandler {
```
Comment (tanner0101) [inline Sources/Vapor/Commands/ServeCommand.swift]: Small nitpick, this `signalSource` gets a suffix where as the previous one doesn't. That could be a little confusing. A better solution here might be to embed a small function that configures the signal source for a given signal.

Comment (tanner0101) [reply Sources/Vapor/Commands/ServeCommand.swift]:
```swift
let signalQueue = DispatchQueue(label: "codes.vapor.server.shutdown")
func makeSignalSource(_ code: Int32) {
    let source = DispatchSource.makeSignalSource(signal: code, queue: signalQueue)
    source.setEventHandler {
            _ = self.runningServer?.close()
            source.cancel()
    }
    source.resume()
    signal(code, SIG_IGN)
}
makeSignalSource(SIGTERM)
makeSignalSource(SIGINT)
```

Comment (heckj) [reply Sources/Vapor/Commands/ServeCommand.swift]: thank you, that makes a huge amount of sense. I'll update this

Comment (heckj) [reply Sources/Vapor/Commands/ServeCommand.swift]: updated PR

**Comment by penny-coin** [comment]
Hey @heckj, you just merged a pull request, have a coin! 

You now have 1 coins.

**Pull request merged**